File: mobile/lib/core/entities/product.dart
import 'entity.dart';
import 'product_option.dart';
import 'product_variant.dart';

class Product implements Entity {
  late final String id;
  late final String title;
  late final String description;
  late final List<Uri> images;
  late final List<ProductVariant> variants;
  late final List<ProductOption> options;

  Product.lazy({
    String? id,
    String? title,
    String? description,
    List<Uri>? images,
    List<ProductVariant>? variants,
    List<ProductOption>? options,
  }) {
    if (id != null) this.id = id;
    if (title != null) this.title = title;
    if (description != null) this.description = description;
    if (images != null) this.images = images;
    if (variants != null) this.variants = variants;
    if (options != null) this.options = options;
  }
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/entity/Product.kt
package me.lucasbatista.vienna.sdk.entity

import java.net.URL

class Product(
    id: String? = null,
    title: String? = null,
    description: String? = null,
    images: List<URL>? = null,
    variants: List<ProductVariant>? = null,
    options: List<ProductOption>? = null,
) {
    lateinit var id: String
    lateinit var title: String
    lateinit var description: String
    lateinit var images: List<URL>
    lateinit var variants: List<ProductVariant>
    lateinit var options: List<ProductOption>

    init {
        id?.let { this.id = it }
        description?.let { this.description = description }
        title?.let { this.title = it }
        images?.let { this.images = it }
        variants?.let { this.variants = variants }
        options?.let { this.options = options }
    }
}